hyperspace/cosmos/src/chain.rs

```diff
@@ -0,0 +1,275 @@
+};
+use prost::Message;
+use std::pin::Pin;
+use tendermint_rpc::{
+	event::{Event, EventData},
+	query::{EventType, Query},
+	SubscriptionClient, WebSocketClient,
+};
+
+#[async_trait::async_trait]
+impl<H> LightClientSync for CosmosClient<H>
+where
+	H: 'static + Clone + Send + Sync,
+{
+	async fn is_synced<C: Chain>(&self, _counterparty: &C) -> Result<bool, anyhow::Error> {
+		Ok(true)
+	}
+
+	async fn fetch_mandatory_updates<C: Chain>(
+		&self,
+		_counterparty: &C,
+	) -> Result<(Vec<Any>, Vec<IbcEvent>), anyhow::Error> {
+		Ok((vec![], vec![]))
+	}
+}
+
+#[async_trait::async_trait]
+impl<H> Chain for CosmosClient<H>
+where
+	H: Clone + Send + Sync + 'static,
+{
+	fn name(&self) -> &str {
+		&*self.name
+	}
+
+	fn block_max_weight(&self) -> u64 {
+		self.max_tx_size as u64
+	}
+
+	async fn estimate_weight(&self, messages: Vec<Any>) -> Result<u64, Self::Error> {
+		let account_info = self.query_account().await?;
+		let fee = self.get_fee();
+		let (_, tx_raw, _) =
+			sign_tx(self.keybase.clone(), self.chain_id.clone(), &account_info, vec![], fee)?;
+
+		let total_len = tx_raw.encoded_len();
+		let body_bytes_len = tx_raw.body_bytes.len();
+		let envelope_len = if body_bytes_len == 0 {
+			total_len
+		} else {
+			total_len - 1 - prost::length_delimiter_len(body_bytes_len) - body_bytes_len
+		};
+		// Full length of the transaction can then be derived from the length of the invariable
+		// envelope and the length of the body field, taking into account the varint encoding
+		// of the body field's length delimiter.
+		fn tx_len(envelope_len: usize, body_len: usize) -> usize {
+			// The caller has at least one message field length added to the body's
+			debug_assert!(body_len != 0);
+			envelope_len + 1 + prost::length_delimiter_len(body_len) + body_len
+		}
+
+		let mut current_count = 0;
+		let mut current_len = body_bytes_len;
+
+		for message in messages {
+			let message_len = message.encoded_len();
+
+			// The total length the message adds to the encoding includes the
+			// field tag (small varint) and the length delimiter.
+			let tagged_len = 1 + prost::length_delimiter_len(message_len) + message_len;
+			if current_count >= 30 ||
+				tx_len(envelope_len, current_len + tagged_len) > self.max_tx_size
+			{
+				return Err(Error::Custom("Too many messages".to_string()))
+			}
+			current_count += 1;
+			current_len += tagged_len;
+		}
+		Ok(current_len as u64)
+	}
+
+	async fn finality_notifications(
+		&self,
+	) -> Pin<Box<dyn Stream<Item = <Self as IbcProvider>::FinalityEvent> + Send + Sync>> {
+		let (ws_client, ws_driver) = WebSocketClient::new(self.websocket_url.clone())
+			.await
+			.map_err(|e| Error::from(format!("Web Socket Client Error {:?}", e)))
+			.unwrap();
+		tokio::spawn(ws_driver.run());
+		let subscription = ws_client
+			.subscribe(Query::from(EventType::NewBlock))
+			.await
+			.map_err(|e| Error::from(format!("failed to subscribe to new blocks {:?}", e)))
+			.unwrap()
+			.chunks(6);
```

Comment:
Here, It seems that `chunks(6)` were added to process every multiple block together since the height containing proofs differs one from the height with the IBC event to allow you to append `UpdateClientMessage`. Right?